### PR TITLE
Init DPOS candidates

### PR DESCRIPTION
**Pull request by pathornteng**
This PR adds logic to register all validators for candidates with minimum whitelist amount. 

Ref: https://github.com/loomnetwork/loomchain/issues/1341

- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

### CODE REVIEW

File: builtin/plugins/dposv3/dpos.go [hunk 1]
```diff
@@ -197,6 +198,40 @@ func (c *DPOS) Init(ctx contract.Context, req *InitRequest) error {
 		params.DowntimePeriod = defaultDowntimePeriod
 	}
 
+	candidates, err := LoadCandidateList(ctx)
```
Comment (enlight) [inline builtin/plugins/dposv3/dpos.go]: Should just be `candidates := &CandidateList{}`

File: builtin/plugins/dposv3/dpos.go [hunk 2]
```diff
@@ -197,6 +198,40 @@ func (c *DPOS) Init(ctx contract.Context, req *InitRequest) error {
...
+	if err != nil {
+		return err
+	}
+
+	// if InitCandidates is true, whitelist validators and register them for candidates
+	if params.InitCandidates {
```
Comment (enlight) [inline builtin/plugins/dposv3/dpos.go]: `params` are persisted in the `State`, there's no reason for this to be in there though.

File: builtin/plugins/dposv3/dpos.go [hunk 3]
```diff
@@ -197,6 +198,40 @@ func (c *DPOS) Init(ctx contract.Context, req *InitRequest) error {
...
+		for i, validator := range req.Validators {
+			candidateAddr := loom.Address{ChainID: "default", Local: loom.LocalAddressFromPublicKey(validator.PubKey)}
```
Comment (enlight) [inline builtin/plugins/dposv3/dpos.go]: Hardcoded chain ID...

File: builtin/plugins/dposv3/dpos.go (hunk 1 above)
Comment (pathornteng) [reply builtin/plugins/dposv3/dpos.go]: Done

File: builtin/plugins/dposv3/dpos.go (hunk 2 above)
Comment (pathornteng) [reply builtin/plugins/dposv3/dpos.go]: Moved it to `DPOSInitRequest` instead

File: builtin/plugins/dposv3/dpos.go (hunk 3 above)
Comment (pathornteng) [reply builtin/plugins/dposv3/dpos.go]: Allow configurable chain ID

File: builtin/plugins/dposv3/dpos.go [hunk 4]
```diff
@@ -197,6 +198,40 @@ func (c *DPOS) Init(ctx contract.Context, req *InitRequest) error {
 		params.DowntimePeriod = defaultDowntimePeriod
 	}
 
+	chainID := "default"
+	if req.ChainId != "" {
+		chainID = req.ChainId
+	}
```
Comment (enlight) [inline builtin/plugins/dposv3/dpos.go]: Should be able to use `ctx.Block().ChainID` here, no?

Comment (pathornteng) [reply builtin/plugins/dposv3/dpos.go]: Done, ChainID was empty in init block but not it is fixed.

**Pull request merged**